link/src/main/java/com/stripe/android/link/ui/LinkButtonView.kt

```diff
@@ -90,39 +86,97 @@ fun LinkButton(
                     bottom = LinkButtonVerticalPadding
                 )
             ) {
-                Icon(
-                    painter = painterResource(R.drawable.stripe_link_logo),
-                    contentDescription = stringResource(StripeR.string.stripe_link),
-                    modifier = Modifier
-                        .height(22.dp)
-                        .padding(
-                            horizontal = 5.dp,
-                            vertical = 3.dp
-                        ),
-                    tint = MaterialTheme.linkColors.buttonLabel
-                        .copy(alpha = LocalContentAlpha.current)
-                )
-                email?.let {
-                    Spacer(modifier = Modifier.weight(1f))
-                    Box(
-                        modifier = Modifier
-                            .background(
-                                color = Color.Black.copy(alpha = 0.05f),
-                                shape = LinkButtonEmailShape
-                            )
-                    ) {
-                        Text(
-                            text = it,
-                            modifier = Modifier
-                                .padding(6.dp),
-                            color = MaterialTheme.linkColors.buttonLabel,
-                            fontSize = 14.sp,
-                            overflow = TextOverflow.Ellipsis,
-                            maxLines = 1
-                        )
-                    }
+                if (email == null) {
+                    SignedOutButtonContent()
+                } else {
+                    SignedInButtonContent(email = email)
                 }
             }
         }
     }
 }
+
+@Composable
+private fun SignedInButtonContent(email: String) {
+    LinkIcon()
+    LinkDivider()
+    LinkAccountInfo(email)
+    LinkArrow()
```

Comment:
nit: I think these should be wrapped in some sort of container

Reply:
The container is inherited from the caller. (All the way in `LinkButton`), but I might be misunderstanding too.

Reply:
+1 on this comment. We generally want to emit a single element from a Composable. If we do want to scope it to a specific type of container, we should do that by making the Composable function an extension method on that scope.

```kotlin
@Composable
private fun RowScope.SignedInButtonContent(email: String) {
    LinkIcon()
    // …
}
```

Here, it might show an “unused receiver” warning on this extension method, so we should suppress that to show that it’s intentional.